File: src/yanzm/example/adaptersamples/SimpleExpandableListAdapterSample.java
package yanzm.example.adaptersamples;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.HashMap;
import java.util.List;

import android.app.ExpandableListActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.ExpandableListView;
import android.widget.SimpleExpandableListAdapter;

public class SimpleExpandableListAdapterSample extends ExpandableListActivity {

	@SuppressWarnings({ "unchecked", "serial" })
	public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        // for group
        final List<HashMap<String, String>> groupData = 
            Arrays.asList(
                new HashMap<String, String>() {{
                    put("Author", "夏目漱石");
                }},
                new HashMap<String, String>() {{
                    put("Author", "太宰治");
                }},
                new HashMap<String, String>() {{
                    put("Author", "夏目漱石");
                }},
                new HashMap<String, String>() {{
                    put("Author", "太宰治");
                }},
                new HashMap<String, String>() {{
                    put("Author", "夏目漱石");
                }},
                new HashMap<String, String>() {{
                    put("Author", "太宰治");
                }},
                new HashMap<String, String>() {{
                    put("Author", "夏目漱石");
                }},
                new HashMap<String, String>() {{
                    put("Author", "太宰治");
                }},
                new HashMap<String, String>() {{
                    put("Author", "夏目漱石");
                }},
                new HashMap<String, String>() {{
                    put("Author", "太宰治");
                }}
            );

        String[] groupFrom = {"Author"};
        
        int[] groupTo = {R.id.textview};

        // for child
        List<List<HashMap<String, String>>> childData = new ArrayList<List<HashMap<String, String>>>();

        final List<HashMap<String, String>> firstChildData = 
            Arrays.asList(
                new HashMap<String, String>() {{
                    put("Title", "我輩は猫である");
                    put("Date", "1905年");
                }},
                new HashMap<String, String>() {{
                    put("Title", "坊ちゃん");
                    put("Date", "1906年");
                }},
                new HashMap<String, String>() {{
                    put("Title", "坊ちゃん");
                    put("Date", "1906年");
                }},
                new HashMap<String, String>() {{
                    put("Title", "坊ちゃん");
                    put("Date", "1906年");
                }},
                new HashMap<String, String>() {{
                    put("Title", "坊ちゃん");
                    put("Date", "1906年");
                }},
                new HashMap<String, String>() {{
                    put("Title", "坊ちゃん");
                    put("Date", "1906年");
                }},
                new HashMap<String, String>() {{
                    put("Title", "坊ちゃん");
                    put("Date", "1906年");
                }},
                new HashMap<String, String>() {{
                    put("Title", "坊ちゃん");
                    put("Date", "1906年");
                }},
                new HashMap<String, String>() {{
                    put("Title", "坊ちゃん");
                    put("Date", "1906年");
                }},
                new HashMap<String, String>() {{
                    put("Title", "坊ちゃん");
                    put("Date", "1906年");
                }},
                new HashMap<String, String>() {{
                    put("Title", "坊ちゃん");
                    put("Date", "1906年");
                }},
                new HashMap<String, String>() {{
                    put("Title", "坊ちゃん");
                    put("Date", "1906年");
                }}
            );
		
		final List<HashMap<String, String>> secondChildData = 
            Arrays.asList(
                new HashMap<String, String>() {{
                    put("Title", "人間失格");
                    put("Date", "1948年");
                }},
                new HashMap<String, String>() {{
                    put("Title", "走れメロス");
                    put("Date", "1940年");
                }}
            );
        
        childData.add(firstChildData);
        childData.add(secondChildData);
        childData.add(firstChildData);
        childData.add(secondChildData);
        childData.add(firstChildData);
        childData.add(secondChildData);
        childData.add(firstChildData);
        childData.add(secondChildData);
        childData.add(firstChildData);
        childData.add(secondChildData);

        String[] childFrom = {"Title", "Date"};
        
        int[] childTo = {R.id.textview1, R.id.textview2};

        ExpandableListView el = getExpandableListView();
        el.setOnGroupClickListener(new ExpandableListView.OnGroupClickListener() {
			
			@Override
			public boolean onGroupClick(ExpandableListView parent, View v,
					int groupPosition, long id) {
				//((ExpandableListView)getExpandableListView()).setSelectionFromTop(groupPosition, 0);
				((ExpandableListView)getExpandableListView()).smoothScrollToPosition(groupPosition);
				//((ExpandableListView)getExpandableListView()).scrollTo(0, groupPosition * 32);
				return false;
			}
		});
        setContentView(el);
        
        SimpleExpandableListAdapter simpleAdapter = new SimpleExpandableListAdapter(this, 
        		groupData, R.layout.list_item6, groupFrom, groupTo,
        		childData, R.layout.list_item7, childFrom, childTo);
        setListAdapter(simpleAdapter);
        
    }

}

File: app/src/main/java/yanzm/example/adaptersamples/SimpleExpandableListAdapterSample.kt
package yanzm.example.adaptersamples

import android.os.Bundle
import android.widget.ExpandableListView
import android.widget.SimpleExpandableListAdapter
import androidx.appcompat.app.AppCompatActivity

class SimpleExpandableListAdapterSample : AppCompatActivity() {

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val listView = ExpandableListView(this)
        setContentView(listView)

        // for group
        val groupData = listOf(
            mapOf("Author" to "夏目漱石"),
            mapOf("Author" to "太宰治"),
            mapOf("Author" to "夏目漱石"),
            mapOf("Author" to "太宰治"),
            mapOf("Author" to "夏目漱石"),
            mapOf("Author" to "太宰治"),
            mapOf("Author" to "夏目漱石"),
            mapOf("Author" to "太宰治"),
            mapOf("Author" to "夏目漱石"),
            mapOf("Author" to "太宰治"),
        )
        val groupFrom = arrayOf("Author")
        val groupTo = intArrayOf(R.id.textview)

        // for child
        val firstChildData = listOf(
            mapOf(
                "Title" to "我輩は猫である",
                "Date" to "1905年",
            ),
            mapOf(
                "Title" to "坊ちゃん",
                "Date" to "1906年",
            ),
            mapOf(
                "Title" to "坊ちゃん",
                "Date" to "1906年",
            ),
            mapOf(
                "Title" to "坊ちゃん",
                "Date" to "1906年",
            ),
            mapOf(
                "Title" to "坊ちゃん",
                "Date" to "1906年",
            ),
            mapOf(
                "Title" to "坊ちゃん",
                "Date" to "1906年",
            ),
            mapOf(
                "Title" to "坊ちゃん",
                "Date" to "1906年",
            ),
            mapOf(
                "Title" to "坊ちゃん",
                "Date" to "1906年",
            ),
            mapOf(
                "Title" to "坊ちゃん",
                "Date" to "1906年",
            ),
            mapOf(
                "Title" to "坊ちゃん",
                "Date" to "1906年",
            ),
            mapOf(
                "Title" to "坊ちゃん",
                "Date" to "1906年",
            ),
            mapOf(
                "Title" to "坊ちゃん",
                "Date" to "1906年",
            ),
        )
        val secondChildData = listOf(
            mapOf(
                "Title" to "人間失格",
                "Date" to "1948年",
            ),
            mapOf(
                "Title" to "走れメロス",
                "Date" to "1940年",
            ),
        )
        val childData = listOf(
            firstChildData,
            secondChildData,
            firstChildData,
            secondChildData,
            firstChildData,
            secondChildData,
            firstChildData,
            secondChildData,
            firstChildData,
            secondChildData,
        )
        val childFrom = arrayOf("Title", "Date")
        val childTo = intArrayOf(R.id.textview1, R.id.textview2)

        listView.setOnGroupClickListener { _, _, groupPosition, _ ->
            listView.smoothScrollToPosition(groupPosition)
            false
        }

        val simpleAdapter = SimpleExpandableListAdapter(
            this,
            groupData, R.layout.list_item6, groupFrom, groupTo,
            childData, R.layout.list_item7, childFrom, childTo
        )

        listView.setAdapter(simpleAdapter)
    }
}
